ThumbnailSequence.cpp

```diff
@@ -1560,7 +1560,7 @@ ThumbnailSequence::CompositeItem::CompositeItem(
 bool
 ThumbnailSequence::CompositeItem::incompleteThumbnail() const
 {
-	return dynamic_cast<IncompleteThumbnail*>(m_pThumb) != 0;
+	return m_pThumb == 0 && dynamic_cast<IncompleteThumbnail*>(m_pThumb) != 0;
```

Comment:
You know, you could just let FreeBSD scantailor maintainer deal with this problem, given that it's a FreeBSD-specific one. The code in question didn't change for a long time.

I would be extremely surprised if the problem turns out to be related to the use of std::auto_ptr. You are free to pursue this course, but it looks like you need to read some articles on auto_ptr and unique_ptr first. I don't really have time to explain every detail, while you can easily find this information on the web.